src/main/main.cpp

```diff
@@ -347,6 +356,48 @@ catchupTo(Config const& cfg, uint32_t to)
     catchup(cfg, to, std::numeric_limits<uint32_t>::max());
 }
 
+static int
+reportLastHistoryCheckpoint(Config const& cfg)
+{
+    VirtualClock clock(VirtualClock::REAL_TIME);
+    Application::pointer app = Application::create(clock, cfg, false);
+
+    if (checkInitialized(app))
+    {
+        app->applyCfgCommands();
+        auto state = HistoryArchiveState{};
+        auto& wm = app->getWorkManager();
+        auto getHistoryArchiveStateWork =
+            wm.addWork<GetHistoryArchiveStateWork>(
+                "get-history-archive-state-work", state);
+        wm.advanceChildren();
+        while (clock.crank(true) && !wm.allChildrenDone())
+            ;
+
+        auto ok = getHistoryArchiveStateWork->getState() == Work::WORK_SUCCESS;
```

Comment:
I think that it's probably better to output this to a file to make parsing by scripts easier.